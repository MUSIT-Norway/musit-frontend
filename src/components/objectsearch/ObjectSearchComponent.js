import React from 'react'
import { I18n } from 'react-i18nify'
import { Grid, Form, FormGroup, FormControl, ControlLabel, Button, Table } from 'react-bootstrap'
import FontAwesome from 'react-fontawesome'
import Breadcrumb from '../../layout/Breadcrumb'
import { createBreadcrumbPath } from '../../util'
import PagingToolbar from '../../util/paging'

export function renderParam(id, props, style) {
  return (
    <FormGroup controlId={id}>
      <ControlLabel>{I18n.t(`musit.objectsearch.${id}.label`)}</ControlLabel>
      {' '}
      <FormControl
        style={{ ...style }}
        type="text"
        placeholder={I18n.t(`musit.objectsearch.${id}.placeHolder`)}
        value={props.params[id] || ''}
        onChange={(e) => props.onChangeField(id, e.target.value)}
      />
    </FormGroup>
  )
}

export function renderBreadcrumb(nodes: []) {
  return <Breadcrumb nodes={nodes} allActive />
}

export default (props) =>
  <div style={{ paddingTop: 20 }}>
    <main>
      <Grid>
        <div>
          <h2>{I18n.t('musit.objectsearch.title')}</h2>
          <Form inline>
            {renderParam('museumNo', props)}
            {' '}
            {renderParam('subNo', props)}
            {' '}
            {renderParam('term', props, { width: '470px' })}
            {' '}
            <Button
              type="submit"
              onClick={(e) => {
                e.preventDefault()
                props.searchForObjects(props.params)
              }}
            >
              <FontAwesome name="search" style={{ fontSize: '1.3em' }} />
            </Button>
          </Form>
          <br />
          <h4>
          {props.loaded &&
            (props.data.totalMatches > 0 ?
              I18n.t('musit.objectsearch.results.title', { count: props.data.totalMatches })
              :
              I18n.t('musit.objectsearch.results.noHit')
            )
          }
          </h4>
          {props.data.matches.length > 0 &&
            <div>
              <PagingToolbar
                numItems={props.data.totalMatches}
                baseUrl={props.location.pathname}
                currentPage={props.params.currentPage}
                perPage={props.params.perPage}
                onClick={(page) => props.searchForObjects(props.params, page)}
              />
              <Table>
                <thead>
                  <tr>
                    <th>{I18n.t('musit.objectsearch.museumNo.label')}</th>
                    <th>{I18n.t('musit.objectsearch.subNo.label')}</th>
                    <th>{I18n.t('musit.objectsearch.term.label')}</th>
                    <th>{I18n.t('musit.objectsearch.location.label')}</th>
                  </tr>
                </thead>
                <tbody>
                {props.data.matches.map((data, i) => {
                  const path = createBreadcrumbPath(data.path, data.pathNames)
                  return (
                    <tr key={i}>
                      <td className="museumNo">{data.museumNo}</td>
                      <td className="subNo">{data.subNo}</td>
                      <td className="term">{data.term}</td>
                      <td className="path">{renderBreadcrumb(path)}</td>
                      <td className="move">
                        <a
                          href=""
                          onClick={(e) => {
                            e.preventDefault()
                            props.pickObject(data, path)
                          }}
                          title={I18n.t('musit.objectsearch.addToPickList')}
                        >
                          <FontAwesome name="shopping-cart" style={{ fontSize: '1.3em' }} />
                        </a>
                      </td>
                    </tr>
                  )
                })}
                </tbody>
              </Table>
              <PagingToolbar
                numItems={props.data.totalMatches}
                baseUrl={props.location.pathname}
                currentPage={props.params.currentPage}
                perPage={props.params.perPage}
                onClick={(page) => props.searchForObjects(props.params, page)}
              />
            </div>
          }
        </div>
      </Grid>
    </main>
  </div>