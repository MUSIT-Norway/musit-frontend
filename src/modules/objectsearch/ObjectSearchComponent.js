import React from 'react';
import {I18n} from 'react-i18nify';
import {Grid, Form, FormGroup, FormControl, ControlLabel, Button, Table} from 'react-bootstrap';
import FontAwesome from 'react-fontawesome';
import Breadcrumb from '../../components/layout/Breadcrumb';
import PagingToolbar from '../../shared/paging';
import {hashHistory} from 'react-router';
import Loader from 'react-loader';


export function renderParam(id, props, style) {
  return (
    <FormGroup controlId={id}>
      <ControlLabel>{I18n.t(`musit.objectsearch.${id}.label`)}</ControlLabel>
      {' '}
      <FormControl
        style={{...style}}
        type="text"
        placeholder={I18n.t(`musit.objectsearch.${id}.placeHolder`)}
        value={props.params[id] || ''}
        onChange={(e) => props.onChangeField(id, e.target.value)}
      />
    </FormGroup>
  );
}

export default (props) =>
  <div style={{paddingTop: 20}}>
    <main>
      <Grid>
        <div>
          <h1>{I18n.t('musit.objectsearch.title')}</h1>
          <Form inline>
            {renderParam('museumNo', props)}
            {' '}
            {renderParam('subNo', props)}
            {' '}
            {renderParam('term', props, {width: '440px'})}
            {' '}
            <Button
              type="submit"
              onClick={(e) => {
                e.preventDefault();
                props.searchForObjects(props.params, 1, props.user.museumId, props.user.collectionId);
              }}
            >
              <FontAwesome name="search" style={{fontSize: '1.3em'}}/>
            </Button>
          </Form>
          <br />
          <h4>
            {props.loaded &&
            (props.data.totalMatches > 0 ?
                I18n.t('musit.objectsearch.results.title', {count: props.data.totalMatches})
                :
                I18n.t('musit.objectsearch.results.noHit')
            )
            }
          </h4>
          <Loader loaded={!props.loading}>
          {props.data.matches.length > 0 &&
          <div>
            <PagingToolbar
              numItems={props.data.totalMatches}
              baseUrl={props.location.pathname}
              currentPage={props.params.currentPage}
              perPage={props.params.perPage}
              onClick={(page) => props.searchForObjects(props.params, page, props.user.museumId, props.user.collectionId)}
            />
            <Table>
              <thead>
              <tr>
                <th>{I18n.t('musit.objectsearch.museumNo.label')}</th>
                <th>{I18n.t('musit.objectsearch.subNo.label')}</th>
                <th>{I18n.t('musit.objectsearch.term.label')}</th>
                <th>{I18n.t('musit.objectsearch.location.label')}</th>
                <th>
                  <a
                    href=""
                    onClick={(e) => {
                      e.preventDefault();
                      props.pickObjects(props.data.matches, props.user.museumId, props.user.collectionId);
                    }}
                    title={I18n.t('musit.objectsearch.addAllToPickList')}
                  >
                    <FontAwesome name="shopping-cart" style={{fontSize: '1.3em'}}/>
                  </a>
                </th>
              </tr>
              </thead>
              <tbody>
                {props.data.matches.map((data, i) => {
                  const isMainObject = !data.mainObjectId || data.isMainObject();
                  const isChildObject = data.mainObjectId && !data.isMainObject();
                  return (
                    <tr key={i} className={isChildObject ? 'childObject' : isMainObject && 'mainObject'}>
                      <td className="museumNo">{data.museumNo}</td>
                      <td className="subNo">{data.subNo}</td>
                      <td className="term">{data.term}</td>
                      <td className="path">
                        {data.breadcrumb.length > 0 &&
                        <Breadcrumb
                          node={data}
                          allActive
                          onClickCrumb={(node) => hashHistory.push(node.url) }
                        />
                        }
                      </td>
                      <td className="move">
                        {isMainObject &&
                        <a
                          href=""
                          onClick={(e) => {
                            e.preventDefault();
                            props.pickObject(data, data.breadcrumb, props.user.museumId, props.user.collectionId);
                          }}
                          title={I18n.t('musit.objectsearch.addToPickList')}
                        >
                          <FontAwesome name="shopping-cart" style={{fontSize: '1.3em'}}/>
                        </a>
                        }
                      </td>
                    </tr>
                  );
                })}
              </tbody>
            </Table>
            <PagingToolbar
              numItems={props.data.totalMatches}
              currentPage={props.params.currentPage}
              perPage={props.params.perPage}
              onClick={(page) => props.searchForObjects(props.params, page, props.user.museumId, props.user.collectionId)}
            />
          </div>
          }
          </Loader>
        </div>
      </Grid>
    </main>
  </div>;