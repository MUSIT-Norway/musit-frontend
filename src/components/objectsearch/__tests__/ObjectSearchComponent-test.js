import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json';
import React from 'react'
import ObjectSearchComponent from '../ObjectSearchComponent'
import { renderBreadcrumb } from '../ObjectSearchComponent'

describe('ObjectSearchComponent', () => {
  const testData = {
    totalMatches: 20,
    matches: Array(20).fill(
      {
        museumNo: '12345',
        subNo: '45',
        term: 'Fuglekasse',
        id: 1,
        path: ',1,2,3,',
        pathNames: [
          {
            nodeId: 1,
            name: "Root",
            type: "Organization"
          },
          {
            nodeId: 2,
            name: "Musit",
            type: "Building"
          },
          {
            nodeId: 3,
            name: "Kontoret",
            type: "Room"
          }
        ]
      }
    )
  }

  it('should display object 1', () => {
    const wrapper = shallow(
      <ObjectSearchComponent
        location={{
          query: {
            page: '1'
          },
          pathname: '/#/search/objects'
        }}
        data={testData}
        onChangeField={() => true}
        params={{
          museumNo: "1234",
          subNo: "455",
          term: "Lololol",
          currentPage: 1
        }}
      />
    )
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('should display breadcrumb correctly', () => {
    const path = ',1,2,3,'
    const pathNames = [
      {
        nodeId: 1,
        name: "Root",
        type: "Organization"
      },
      {
        nodeId: 2,
        name: "Musit",
        type: "Building"
      },
      {
        nodeId: 3,
        name: "Kontoret",
        type: "Room"
      }
    ]
    const wrapper = shallow(renderBreadcrumb(path, pathNames))
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
