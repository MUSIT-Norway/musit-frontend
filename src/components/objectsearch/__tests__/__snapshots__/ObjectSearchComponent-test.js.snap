exports[`ObjectSearchComponent should display object 1 1`] = `
<div
  style={
    Object {
      "paddingTop": 20
    }
  }>
  <main>
    <Grid
      bsClass="container"
      componentClass="div"
      fluid={false}>
      <div>
        <h2>
          Title
        </h2>
        <Form
          bsClass="form"
          componentClass="form"
          horizontal={false}
          inline={true}>
          <FormGroup
            bsClass="form-group"
            controlId="formInlineName">
            <ControlLabel
              bsClass="control-label"
              srOnly={false}>
              Museumnr.
            </ControlLabel>
             
            <FormControl
              bsClass="form-control"
              componentClass="input"
              placeholder=""
              type="text" />
          </FormGroup>
           
          <FormGroup
            bsClass="form-group"
            controlId="formInlineName">
            <ControlLabel
              bsClass="control-label"
              srOnly={false}>
              Unr.
            </ControlLabel>
             
            <FormControl
              bsClass="form-control"
              componentClass="input"
              placeholder=""
              type="text" />
          </FormGroup>
           
          <FormGroup
            bsClass="form-group"
            controlId="formInlineName">
            <ControlLabel
              bsClass="control-label"
              srOnly={false}>
              Term/Artsnavn
            </ControlLabel>
             
            <FormControl
              bsClass="form-control"
              componentClass="input"
              placeholder=""
              type="text" />
          </FormGroup>
           
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="default"
            disabled={false}
            type="submit">
            Search
          </Button>
        </Form>
        <div>
          <br />
          <h4>
            Resultat - s√∏ket ga x treff:
          </h4>
          <Table
            bordered={false}
            bsClass="table"
            condensed={false}
            hover={false}
            responsive={false}
            striped={false}>
            <thead>
              <th>
                Museumnr.
              </th>
              <th>
                Unr.
              </th>
              <th>
                Term/Artsnavn
              </th>
              <th>
                Plassering
              </th>
              <th />
            </thead>
            <tbody>
              <tr>
                <td
                  className="museumNo">
                  12345
                </td>
                <td
                  className="subNo">
                  45
                </td>
                <td
                  className="term">
                  Fuglekasse
                </td>
                <td
                  className="path">
                  / Museum / Bygg 1 / Rommet
                </td>
                <td
                  className="move">
                  <FontAwesome
                    name="truck" />
                </td>
              </tr>
              <tr>
                <td
                  className="museumNo">
                  12
                </td>
                <td
                  className="subNo">
                  46
                </td>
                <td
                  className="term">
                  Stol
                </td>
                <td
                  className="path">
                  / Museum / Bygg 1 / Rommet
                </td>
                <td
                  className="move">
                  <FontAwesome
                    name="truck" />
                </td>
              </tr>
              <tr>
                <td
                  className="museumNo">
                  123
                </td>
                <td
                  className="subNo">
                  48
                </td>
                <td
                  className="term">
                  Kasse
                </td>
                <td
                  className="path">
                  / Museum / Bygg 1 / Rommet
                </td>
                <td
                  className="move">
                  <FontAwesome
                    name="truck" />
                </td>
              </tr>
            </tbody>
          </Table>
          <span
            style={
              Object {
                "float": "right"
              }
            }>
            1 2 3 Neste side
          </span>
        </div>
      </div>
    </Grid>
  </main>
</div>
`;
